stm32: uart: fix non-LP UARTs with PM

mickael868 opened the pull request:
When the system goes to STOP2 mode, non-LP USARTs loose their configuration. This patch tests if the USART Enable bit is reset on wake-up, and if so, it re-configures the registers.

Related to [this issue](https://github.com/zephyrproject-rtos/zephyr/issues/94036).

Tested on a STM32WLE5 target.

Also fix a typo.

etienne-lms commented on drivers/serial/uart_stm32.c:
```diff
@@ -2333,6 +2333,14 @@ static int uart_stm32_pm_action(const struct device *dev,
 			 */
 			uart_stm32_init(dev);
 		}
+
+		/* Non-LP UART peripherals loose their registers in STOP2 mode. */
+		else if (!LL_USART_IsEnabled(config->usart)) {
```
Prefer:
```suggestion
		} else if (!LL_USART_IsEnabled(config->usart)) {
			/* Non-LP UART peripherals loose their registers in STOP2 mode */

mickael868 replied:
Thanks, done.

etienne-lms reviewed (commented):
Could you update the commit header line to a more standard one:
```patch
-stm32: uart: fix non-LP UARTs with PM
+drivers: serial: stm32: fix non-LP UARTs with PM
```

mickael868 commented:
The `if ()` block I added is similar to the existing block right above. Maybe we could merge both with something like:

```c

if (!LL_USART_IsEnabled(config->usart)){
    /* When exiting low power mode, check whether UART is enabled.
     * If not, it means that the peripheral configuration has been erased
     * by the low power mode.
     * Either do a full re-initialization after a Suspend to RAM mode (STM32
     * Standby), or simply configure the registers after a STOP2 mode.
     */
    if (IS_ENABLED(CONFIG_PM_S2RAM)){
        err = uart_stm32_init(dev);
    }

    else{
        err = uart_stm32_registers_configure(dev);
    }

    if (err < 0)
    {
        return err;
    }
}
```

erwango commented:
> The if () block I added is similar to the existing block right above. Maybe we could merge both with something like:

That would be cleaner indeed, thanks.

mathieuchopstm commented on drivers/serial/uart_stm32.c:
```diff
@@ -2318,20 +2316,28 @@ static int uart_stm32_pm_action(const struct device *dev,
 			/* When exiting low power mode, check whether UART is enabled.
-			 * If not, it means we are exiting Suspend to RAM mode (STM32
-			 * Standby), and the driver needs to be reinitialized.
+			 * If not, it means that the peripheral configuration has been erased
+			 * by the low power mode.
+			 * Either do a full re-initialization after a Suspend to RAM mode (STM32
+			 * Standby), or simply configure the registers after a STOP2 mode.
```
I would prefer a more explicit documentation. Looking at STM32WL5 RefMan (+ some unrelated rewording), I propose:
```suggestion
			 * If not, it means the peripheral has been powered down
			 * by the low-power mode. If suspend-to-RAM is enabled,
			 * assume the entire SoC has been powered down and do a
			 * full re-initialization. Otherwise, assume that the
			 * low-power mode shut down power to the UART but not
			 * critical peripherals (CPU, GPIO, RCC), which means
			 * we only have to reconfigure this UART instance.
			 *
			 * STOP2 on STM32WLE5 is an example of such low-power mode.
```

Commit:
drivers: serial: stm32: fix non-LP UARTs with PM

Re-configure the registers if they re erased because of a STOP2 low power
mode.

Return the error code when re-initializing the UART after a STANDBY low
power mode.

Fix few typos and comments.

Run clang format on uart_stm32_pm_action()

Signed-off-by: Mickael Bosch <mickael.bosch@linux.com>

erwango reviewed (approved):
Non-blocking: Please avoid mixing functional changes and reformating in the same commit (next time).
